src/odemis/driver/test/xt_client_test.py

```diff
@@ -1170,6 +1122,50 @@ def test_beam_powerVA(self):
         self.assertTrue(self.scanner.power.value)
         self.scanner.power.value = init_beam_power
 
+    def test_apply_autostigmator(self):
+        """
+        Test for the auto stigmation functionality.
+        """
+        detector = 'se-detector'  # autostigmator only works for the se-detector with channel=electron1
+        channel = DETECTOR2CHANNELNAME[detector]
+
+        # Run autostigmator flash script
+        autostigmator_future = self.scanner.applyAutoStigmator(detector)
+        # Don't check .is_autostigmating, function uses the same lock as .applyAutoStigmator, so it's only
+        # ever called after the autoStigmator function finished. TODO: is that intended?
```

Comment:
It is known that all functions use the same lock, so that if you have a blocking call you cannot call anything else in parallel. It is hard to work around this. If you are interested: in `XTTKDetector.stop_generate()` there is a workaround, to be able to stop an image acquisition. 

Also `is_autostigmating` does not work for the flash functions. I think we should just remove `is_autostigmating` and `set_autostigmator`, since the `set_autostigmator` is not good enough in autostigamting.

Reply:
Indeed, I thought that the is_autostigmating function wouldn't work for the flash function. Do you really want to delete it? It's not going to be used anymore? What about the set_autofocus? Delete also?

Reply:
We will never use set_autostigmator and is_autostigmating, so it's safe to delete them. 

set_autofocus also has an xtlib implementation, so we should not delete that. Systems without xttoolkit can use that function, instead of the flash function.